source/Halibut.Tests/Support/BackwardsCompatibility/HalibutTestBinaryRunner.cs

```diff
@@ -143,10 +171,13 @@ await Cli.Wrap(new HalibutTestBinaryPath().BinPath(version))
                 }
             }, cancellationToken);
 
-            await Task.WhenAny(runningTentacle, hasTentacleStarted.WaitAsync(cancellationToken), Task.Delay(TimeSpan.FromMinutes(1), cancellationToken));
+            var completedTask = await Task.WhenAny(runningTentacle, hasTentacleStarted.WaitAsync(), Task.Delay(TimeSpan.FromMinutes(1), cancellationToken));
 
             // Will throw.
-            if (runningTentacle.IsCompleted) await runningTentacle;
+            if (completedTask == runningTentacle)
```

Comment:
I think checking Is completed may be inaccurate sometimes due to a race condition. I fixed issues in tentacle client caused by using IsCompleted which reported false while it was the task that caused When any to exit!!